django/db/models/fields/related_descriptors.py

```diff
@@ -193,14 +193,7 @@ def __set__(self, instance, value):
         - ``instance`` is the ``child`` instance
         - ``value`` in the ``parent`` instance on the right of the equal sign
         """
-        # If null=True, we can assign null here, but otherwise the value needs
-        # to be an instance of the related class.
-        if value is None and self.field.null is False:
```

Comment:
I described this change in a more detailed way here:

https://code.djangoproject.com/ticket/25349#comment:6